### PR TITLE
Add VS2019 Support for C# Debugging (c_sharp_basics.rst)

### PR DESCRIPTION
**Pull request by jasonswearingen**
Add docs showing how VS2019 is supported.   re: https://github.com/godotengine/godot-proposals/issues/371

### CODE REVIEW

**Comment by neikeq** [comment]
I've never used this so I can't really review this. However, our VS extension is coming this month, so there's not much of a point in this. It could be useful in the docs for older versions though.

**Comment by jasonswearingen** [comment]
@neikeq is there somewhere this and other C# related roadmaps are documented?  I wouldnt have spent the (considerable) time getting this working if I had known.

**Comment by neikeq** [comment]
I mentioned it in the last two progress reports on the Godot devlog. But in any case, this is still useful for older versions like 3.1.x which a lot of people use.
The changes required in Godot to support the VS extension are minimal, so I would like to include it in 3.2.0. It may be rejected though since the stable release is imminent, in which case this would also be useful for 3.2.0.

File: getting_started/scripting/c_sharp/c_sharp_basics.rst [hunk 1]
```diff
@@ -265,3 +270,18 @@ Profiling your C# code
 
 - `Mono log profiler <https://www.mono-project.com/docs/debug+profile/profile/profiler/>`_ is available via `this PR <https://github.com/godotengine/godot/pull/26513>`_
 - External Mono profiler like `JetBrains dotTrace <https://www.jetbrains.com/profiler/>`_ can be used as described `here <https://github.com/godotengine/godot/pull/34382>`_.
+
+Configure VS2019 for Debugging
+-------------------------------
+1. Install VS2019 with ```.NET desktop development``` and ```Desktop development with C++``` workloads installed.
+2.  **Ensure that you do not have ```Xamarin``` installed** (do not choose ```Mobile development with .NET``` workload.   Xamarin changes what dll's MonoDebugger uses, which breaks debugging.
```
Comment (Calinou) [inline getting_started/scripting/c_sharp/c_sharp_basics.rst]:
```suggestion
2.  **Ensure that you do not have ```Xamarin``` installed** (do not choose ```Mobile development with .NET``` workload.  Xamarin changes the DLLs used by MonoDebugger, which breaks debugging.
```

File: getting_started/scripting/c_sharp/c_sharp_basics.rst [hunk 2]
```diff
@@ -265,3 +270,18 @@ Profiling your C# code
 
 - `Mono log profiler <https://www.mono-project.com/docs/debug+profile/profile/profiler/>`_ is available via `this PR <https://github.com/godotengine/godot/pull/26513>`_
 - External Mono profiler like `JetBrains dotTrace <https://www.jetbrains.com/profiler/>`_ can be used as described `here <https://github.com/godotengine/godot/pull/34382>`_.
+
+Configure VS2019 for Debugging
+-------------------------------
```
Comment (Calinou) [inline getting_started/scripting/c_sharp/c_sharp_basics.rst]: Add a blank line after the header:

```suggestion
-------------------------------

```

File: getting_started/scripting/c_sharp/c_sharp_basics.rst [hunk 3]
```diff
@@ -265,3 +270,18 @@ Profiling your C# code
...
+1. Install VS2019 with ```.NET desktop development``` and ```Desktop development with C++``` workloads installed.
+2.  **Ensure that you do not have ```Xamarin``` installed** (do not choose ```Mobile development with .NET``` workload.   Xamarin changes what dll's MonoDebugger uses, which breaks debugging.
+3. Install the `VSMonoDebugger extension <https://marketplace.visualstudio.com/items?itemName=GordianDotNet.VSMonoDebugger0d62>`_
+4, In VS2019 --> Extensions --> Mono --> Settings:
+  - Select ```Debug/Deploy to local Windows```
+  - Leave ```Local Deploy Path``` blank
+  - Set the ```Mono Debug Port``` to the port in Godot --> Project --> Project Settings --> Mono --> Debugger Agent
+    - Also select ```Wait for Debugger``` in the Godot Mono options.  `This Godot Addon <https://godotengine.org/asset-library/asset/435>`_ may be helpful.
+5. Run the game in Godot.  It should hang at the Godot splash screen while it waits for your debugger to attach.
+6. In VS2019, open your project and choose Extensions -->Mono --> Attach to mono Debugger.
```
Comment (Calinou) [inline getting_started/scripting/c_sharp/c_sharp_basics.rst]:
```suggestion
6. In VS2019, open your project and choose Extensions -->Mono --> Attach to Mono Debugger.
```

Comment (jasonswearingen) [reply getting_started/scripting/c_sharp/c_sharp_basics.rst]: the actual item in the menu has "mono" non-capitalized, but okay :)

File: getting_started/scripting/c_sharp/c_sharp_basics.rst [hunk 4]
```diff
@@ -265,3 +270,17 @@ Profiling your C# code
...
+
+1. Install VS2019 with ```.NET desktop development``` and ```Desktop development with C++``` workloads installed.
+2.  **Ensure that you do not have ```Xamarin``` installed** (do not choose ```Mobile development with .NET``` workload.  Xamarin changes the DLLs used by MonoDebugger, which breaks debugging.
+3. Install the `VSMonoDebugger extension <https://marketplace.visualstudio.com/items?itemName=GordianDotNet.VSMonoDebugger0d62>`_
+4, In VS2019 --> Extensions --> Mono --> Settings:
+  - Select ```Debug/Deploy to local Windows```
+  - Leave ```Local Deploy Path``` blank
+  - Set the ```Mono Debug Port``` to the port in Godot --> Project --> Project Settings --> Mono --> Debugger Agent
+    - Also select ```Wait for Debugger``` in the Godot Mono options.  `This Godot Addon <https://godotengine.org/asset-library/asset/435>`_ may be helpful.
+5. Run the game in Godot.  It should hang at the Godot splash screen while it waits for your debugger to attach.
```
Comment (Calinou) [inline getting_started/scripting/c_sharp/c_sharp_basics.rst]:
```suggestion
5. Run the game in Godot. It should hang at the Godot splash screen while it waits for your debugger to attach.
```

File: getting_started/scripting/c_sharp/c_sharp_basics.rst [hunk 5]
```diff
@@ -265,3 +270,17 @@ Profiling your C# code
...
+6. In VS2019, open your project and choose Extensions -->Mono --> Attach to Mono Debugger.
```
Comment (Calinou) [inline getting_started/scripting/c_sharp/c_sharp_basics.rst]: Sorry, I didn't notice that typo on the first review (or it wasn't there previously):

```suggestion
6. In VS2019, open your project and choose Extensions --> Mono --> Attach to Mono Debugger.
```

**Comment by Calinou** [comment]
Thanks!

**Pull request merged**